src/api/spec/models/comment_spec.rb

```diff
@@ -0,0 +1,86 @@
+require "rails_helper"
+
+RSpec.shared_examples "a comment" do
+  let(:comment_type) { described_class.name.underscore }
+  let(:comment) { create(comment_type) }
+
+  describe "A comment" do
+    it { expect(build(comment_type)).to be_valid }
+  end
+
+  describe "Comment associations" do
+    it { is_expected.to belong_to(:bs_request).inverse_of(:comments) }
+    it { is_expected.to belong_to(:project).inverse_of(:comments) }
+    it { is_expected.to belong_to(:package).inverse_of(:comments) }
+    it { is_expected.to belong_to(:user).inverse_of(:comments) }
+
+    it { is_expected.to have_many(:children) }
+  end
+
+  describe "Comment validations" do
+    it { is_expected.to validate_presence_of(:body) }
+    it { is_expected.to validate_presence_of(:type) }
+    it { is_expected.to validate_presence_of(:user) }
+  end
+
+  describe "to_xml" do
+    context "without parent" do
+      let(:builder) {
+        builder = Nokogiri::XML::Builder.new
+        comment.to_xml(builder)
+        builder
+      }
+
+      context "returns xml" do
+        context "comment element" do
+          let(:comment_element) { builder.doc.css('comment') }
+
+          it "comment attributes and content" do
+            expect(comment_element.attribute('id').value).to match(/^\d+$/)
+            expect(comment_element.attribute('when').value).to match(/^\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d UTC$/)
+            expect(comment_element.attribute('who').value).to match(/^user_\d+$/)
+
+            expect(comment_element.text).to match(/^#<Nokogiri::XML::Builder::NodeBuilder:0x\h+>$/)
+          end
+        end
+      end
+    end
+  end
+
+  describe "blank_or_destroy" do
+    context "without children" do
+      before do
+        comment
+      end
+
+      it 'should be destroyed' do
+        expect { comment.blank_or_destroy }.to change { Comment.count }.by(-1)
+      end
+    end
+
+    context "with children" do
+      before do
+        create(comment_type, parent: comment)
+      end
+
+      it 'should be destroyed' do
+        expect { comment.blank_or_destroy }.to_not change { Comment.count }
+
+        expect(comment.body).to eq 'This comment has been deleted'
+        expect(comment.user).to eq User.find_nobody!
```

Comment:
I would test check the user login instead of the whole User to ensure that the find_nobody method is working properly:

`expect(comment.user.login).to eq '_nobody_'`